File: public/docs/_examples/tutorial/ts/app/hero.service.ts
import { Injectable } from '@angular/core';

import { HEROES }     from './mock-heroes';

@Injectable()
export class HeroService {
  getHeroes() {
    return Promise.resolve(HEROES);
  }

	getHero(id: number) {
    return Promise.resolve(HEROES).then(
      heroes => heroes.filter(hero => hero.id === id)[0]
    );
  }
}

File: public/docs/_examples/tutorial/dart/lib/hero_service.dart
import 'dart:async';

import 'package:angular2/angular2.dart';

import 'hero.dart';
import 'mock_heroes.dart';

@Injectable()
class HeroService {
  Future<List<Hero>> getHeroes() => new Future(() => mockHeroes);

  Future<List<Hero>> getHeroesSlowly() {
    return new Future.delayed(const Duration(seconds: 2), () => mockHeroes);
  }

	Future<Hero> getHero(int id) async {
    List<Hero> heroes = await getHeroes();
    return heroes.firstWhere((h) => h.id == id);
	}
}

